hbase-oss/src/main/java/org/apache/hadoop/hbase/oss/HBaseObjectStoreSemantics.java

```diff
@@ -170,6 +173,26 @@ public FSDataInputStream open(Path f) throws IOException {
     }
   }
 
+  @Override
+  public FutureDataInputStreamBuilder openFile(final Path path)
+      throws IOException, UnsupportedOperationException {
+    return new LockedFutureDataInputStreamBuilder(sync, path,
+        fs.openFile(path));
+  }
+
+  /**
+   * This is mostly unsupported, and as there's no way to
+   * get the path from a pathHandle, impossible to lock.
+   * @param pathHandle path
+   * @return never returns successfully.
+   * @throws UnsupportedOperationException always
+   */
+  @Override
+  public FutureDataInputStreamBuilder openFile(final PathHandle pathHandle)
+      throws IOException, UnsupportedOperationException {
+    throw new UnsupportedOperationException("openFile(PathHandle) unsupported");
+  }
+
```

Comment:
It seems we also missed wrapping `open(PathHandle fd, int bufferSize)`, on HBaseObjectStoreSemantics. Can we do something similar?

Reply:
yes, just reject it. you are currently never going to be trying to use hboss on a store which implements it (hdfs, webhdfs), but due diligence is always good